src/linear_algebra/special_linear_group.lean

```diff
@@ -166,6 +176,33 @@ by { ext v i, rw [coe_to_GL, to_lin'_mul], refl }
 def embedding_GL : (special_linear_group n R) →* (general_linear_group R (n → R)) :=
 ⟨λ A, to_GL A, by simp, by simp⟩
 
+
+section has_neg
+
+/-- Formal operation of negation on special linear group on even cardinality `n` given by negating
+each element. -/
+instance {R : Type*} [comm_ring R] [_i : fact (even (fintype.card n))] :
+  has_neg (special_linear_group n R) :=
+⟨λ g, ⟨- g.1, begin
+  convert @det_smul _ _ _ _ _ g (-1),
+  { simp,
+    refl },
+  { convert (det_coe_fun g).symm,
+    simp [nat.neg_one_pow_of_even _i.elim] }
+end⟩⟩
+
+@[simp] lemma has_neg_coe_mat {R : Type*} [comm_ring R]
+  (g : (special_linear_group n R)) [_i : fact (even (fintype.card n))] :
+  @coe _ (matrix n n R) _ (- g) = - (@coe _ (matrix n n R) _ g) :=
+rfl
+
+@[simp] lemma has_neg_coe_fn {R : Type*} [comm_ring R]
+  (g : (special_linear_group n R)) [_i : fact (even (fintype.card n))] :
+  @coe_fn _ _ (- g) = - (@coe_fn _ _ g) :=
+rfl
```

Comment:
This messiness (which I understand is not your fault, and is copied from further up in the file) inspired me to PR #8569 - are you ok with waiting for that to be merged first?

Reply:
This lemma is no longer needed, and the `simp_nf` lemma should reject it.
```suggestion
```